### PR TITLE
operators: Document heuristic annotation pullspecs

### PR DESCRIPTION
**Pull request by chmeliik**
* CLOUDBLD-907

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

### CODE REVIEW

File: docs/users.rst [hunk 1]
```diff
@@ -1408,12 +1408,40 @@ Before OSBS can pin your pullspecs, it first needs to find them. Because
...
+
+3. All initContainers in each deployment
+
+   jq: ``.spec.install.spec.deployments[].spec.template.spec.initContainers[]``
+
+4. All RELATED\_IMAGE\_* variables for all containers and initContainers
+
+   jq: ``.env[] | select(.name | test("RELATED_IMAGE_"))`` for each of [2], [3]
+
+5. OSBS tries to do the (almost) impossible and find all pullspecs in all
```
Comment (athos-ribeiro) [inline docs/users.rst:1427]: Could we just say that we are applying some heuristics here? That should imply it is prone to error.

File: docs/users.rst [hunk 2]
```diff
@@ -1408,12 +1408,40 @@ Before OSBS can pin your pullspecs, it first needs to find them. Because
...
+   annotations. See *heuristic annotations* below.
+
+**Heuristic annotations**
+
+OSBS will attempt to extract all pullspecs from all attributes of each
+``metadata.annotations`` object in a CSV. If an attribute contains more than
+one pullspec (as text, e.g. comma-separated), all of them should be found.
+How OSBS does that is not important for the purposes of this document, but an
```
Comment (athos-ribeiro) [inline docs/users.rst:1435]: Here we should either point to the code performing the heuristics (with a PR or commit reference) or mention the "how" for the heuristic.

File: docs/users.rst [hunk 3]
```diff
@@ -1408,12 +1408,40 @@ Before OSBS can pin your pullspecs, it first needs to find them. Because
...
+important thing to note is that only pullspecs conforming to a relatively
+strict format will be found this way:
+
+``registry/namespace*/repo:tag`` or ``registry/namespace*/repo@sha256:digest``
+
+Any number of namespaces, including 0, is valid. Registry must contain at least
+one dot: ``registry.io`` is valid but ``localhost`` is not. Digest, if present,
+must be exactly 64 base16 characters. Tag or digest *must* be specified,
+implicit ``latest`` is not supported.
```
Comment (athos-ribeiro) [inline docs/users.rst:1444]: Can we add "tag"  after latest here?

**Comment by MartinBasti** [comment]
Please address it in separate PR, I urgently need this merged :)

**Pull request merged**